cmd/site/integration_test.go

```diff
@@ -0,0 +1,266 @@
+			Relation("Tracks").
+			Select(),
+	)
+
+	assert.Equal(c.t, len(*expectedPlaylists), len(playlists))
+
+	for i, playlist := range playlists {
+		expectedPlaylist := (*expectedPlaylists)[i]
+		assert.Equal(c.t, playlist.SPM, expectedPlaylist.SPM)
+		// we manually set the id of a new playlist in our mock spotify service
+		// to be "IDFOR:" + the provided name
+		assert.Equal(c.t, "IDFOR:"+expectedPlaylist.Name(), playlist.SpotifyID)
+
+		assert.Equal(c.t, len(expectedPlaylist.Tracks), len(playlist.Tracks))
+		for i, track := range playlist.Tracks {
+			expectedTrack := expectedPlaylist.Tracks[i]
+
+			assert.Equal(c.t, expectedTrack.SpotifyID, track.SpotifyID)
+			assert.Equal(c.t, expectedTrack.Status, track.Status)
+		}
+	}
+}
+
+func (c *cuke) theFollowingSpotifyTracksExist(tracks *[]spotify.AnalyzedTrack) {
+	c.mockSpotify.AddTracks(*tracks)
+}
+
+func (c *cuke) theSpotifyUserHasTheFollowingTracks(userID string, trackIDs []string) {
+	assert.NoError(c.t, c.mockSpotify.AddUserTracks(userID, trackIDs))
+}
+
+func (c *cuke) theSpotifyUserHasTheFollowingPlaylists(userID string, expectedPlaylists *[]internal.Playlist) {
+	user, ok := c.mockSpotify.User(userID)
+	assert.True(c.t, ok)
+
+	assert.Equal(c.t, len(user.Playlists), len(*expectedPlaylists))
+	for i, playlist := range user.Playlists {
+		expectedPlaylist := (*expectedPlaylists)[i]
+
+		assert.Equal(c.t, expectedPlaylist.Name(), playlist.Name)
+
+		assert.Equal(c.t, len(playlist.Tracks), len(expectedPlaylist.Tracks))
+		for i, track := range playlist.Tracks {
+			expectedTrack := expectedPlaylist.Tracks[i]
+
+			assert.Equal(c.t, expectedTrack.SpotifyID, track.ID)
+		}
+	}
+}
+
+func (c *cuke) given() *cuke {
+	return c
+}
+func (c *cuke) when() *cuke {
+	return c
+}
+func (c *cuke) then() *cuke {
+	return c
+}
+func (c *cuke) and() *cuke {
+	return c
+}
+
+func TestLibrarySync(t *testing.T) {
+	// spotify tracks
+	var (
+		// bebey - theopilus london
+		bebey = spotify.AnalyzedTrack{Track: spotify.Track{ID: "7EZ8HvyKfad2vUIOivqjN5"}, Tempo: 164.044}
+		// cheat code - dap the contract
+		cheatCode = spotify.AnalyzedTrack{Track: spotify.Track{ID: "4buERRuDerqKP2g0GAOg4V"}, Tempo: 166.084}
+		// marilyn - mount kimbie
+		marilyn = spotify.AnalyzedTrack{Track: spotify.Track{ID: "5jJPcImQkogKdwsVS36zH7"}, Tempo: 179.826}
+		// if you call - angie mcmahon
+		ifYouCall = spotify.AnalyzedTrack{Track: spotify.Track{ID: "7J29qM3iwzKv83ZnTLdZ0v"}, Tempo: 106.704}
+	)
+	// spotify userIDs
+	var (
+		zach        = "zach"
+		strideSongs = "stridesongs"
+	)
+	cuke := cuke{t: t}
+	after, err := cuke.before()
+	assert.NoError(t, err)
+	defer after()
+
+	background := func() {
+		cuke.given().theFollowSpotifyUsersExist([]string{strideSongs})
+	}
+
+	// note: at the moment cuke doesn't take the new `t *testing.T` on a t.Run...
+	// i'm not sure at the moment since we only have one test so for now i'm
+	// gonna leave it.
+	t.Run("user syncs their library with stride songs", func(t *testing.T) {
```

Comment:
At some point should test some of the spotify pagination stuff, like if a user has 125 tracks with the SPM 150